spyder/plugins/run/plugin.py

```diff
@@ -422,15 +424,16 @@ def create_run_button(
         extra_action_name: Optional[str]
             The name of the action to execute on the run input provider
             after requesting the run input.
-        conjunction_or_preposition: str
-            The conjunction or preposition used to describe the action that
-            should take place after the context, e.g. run <and> advance,
-            run selection <from> the current line, etc. Default: "and".
-        add_to_toolbar: bool
+        context_modificator: Optional[str]
+            The name of the modification to apply to the action.
+            e.g. run selection <up to line>
+        add_to_toolbar: object
             If True, then the action will be added to the Run section of the
-            main toolbar.
-        add_to_menu: bool
+            main toolbar. If a string, it will be a toolbat id
+        add_to_menu: object
             If True, then the action will be added to the Run menu.
+            If a tuple of 3 strings, it corresponds to 
+            (menu_id, menu_section, before_section)
```

Comment:
Please change this to be a dictionary with the following entries:

```python
{'menu': ..., 'section': ..., 'before_section': ...}
```

That way is much easier to understand how this keyword works.